### PR TITLE
Adds support for free google play users.

### PR DESCRIPTION
**Pull request by fergyfresh**
This PR Cleans up @TheModdersDen PR #165 I will take the time to clean up based on @zhaostu's feedback in order to get this in to allow us to close out the majority of the issues which are Free Music related.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# General Python files
 *.pyc
 .venv
 .env
+
+# Visual Studio 2017 files
```
Comment (stevenleeg) [inline .gitignore]: 🙅‍♂️ kill all visual studio stuff here

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# General Python files
```
Comment (stevenleeg) [inline .gitignore]: remove

File: geemusic/__init__.py [hunk 3]
```diff
@@ -10,10 +10,15 @@
 app = Flask(__name__)
 ask = Ask(app, '/alexa')
 
+def lambda_handler(event, _context):
+    return ask.run_aws_lambda(event)
+
 if str(environ['DEBUG_MODE']) is True:
-    log_level = logging.DEBUG
+	log_level = logging.DEBUG
```
Comment (stevenleeg) [inline geemusic/__init__.py]: switch tabs to 4 spaces (might want to do this throughout the entire PR)

File: geemusic/intents/selection.py [hunk 4]
```diff
@@ -39,20 +40,30 @@ def play_artist(artist_name):
...
 
-    # Setup the queue
-    first_song_id = queue.reset(artist['topTracks'])
+    #api.get_artist found somthing from store api search 
+    if 'topTracks' in artist:
+        # Setup the queue
+        first_song_id = queue.reset(artist['topTracks'])
+        name = 'name'
+        track_type = 'top'
+    else: #api.get_artist searched the uploaded library instead, artist is a list
```
Comment (stevenleeg) [inline geemusic/intents/selection.py]: move comment to the next line

File: geemusic/intents/selection.py [hunk 5]
```diff
@@ -39,20 +40,30 @@ def play_artist(artist_name):
...
+    thumbnail = api.get_thumbnail(artist)
+    speech_text = "Playing %s tracks by %s" % (track_type, artist[name])
+
     return audio(speech_text).play(stream_url) \
         .standard_card(title=speech_text,
-                       text='',
-                       small_image_url=thumbnail,
-                       large_image_url=thumbnail)
-
+                        text='',
```
Comment (stevenleeg) [inline geemusic/intents/selection.py]: fix spacing

File: geemusic/intents/selection.py [hunk 6]
```diff
@@ -61,18 +72,25 @@ def play_album(album_name, artist_name):
     # Fetch the album
     album = api.get_album(album_name, artist_name)
 
+    #store dict
+    if type(album) is dict:
+        # Setup the queue
+        first_song_id = queue.reset(sorted(album, key=lambda record: record['trackNumber']))
+        name = 'name'
+    else: #library list
```
Comment (stevenleeg) [inline geemusic/intents/selection.py]: comment on next line

File: geemusic/intents/selection.py [hunk 7]
```diff
@@ -117,77 +135,7 @@ def play_song(song_name, artist_name):
...
-    speech_text = "Playing %s by %s" % (song['title'], song['artist'])
-    return audio(speech_text).play(stream_url) \
-        .standard_card(title=speech_text,
-                       text='',
-                       small_image_url=thumbnail,
-                       large_image_url=thumbnail)
-
-
-@ask.intent("GeeMusicPlaySimilarSongsRadioIntent")
-def play_similar_song_radio():
```
Comment (stevenleeg) [inline geemusic/intents/selection.py]: why is this function removed?

File: geemusic/intents/selection.py [hunk 8]
```diff
@@ -197,27 +145,27 @@ def play_song_radio(song_name, artist_name, album_name):
...
-
-    # Get a streaming URL for the top song
     stream_url = api.get_stream_url(first_song_id)
 
-    thumbnail = api.get_thumbnail(artist['artistArtRef'])
+    thumbnail = api.get_thumbnail(artist)
     speech_text = "Playing %s radio" % artist['name']
     return audio(speech_text).play(stream_url) \
-        .standard_card(title=speech_text,
+          .standard_card(title=speech_text,
```
Comment (stevenleeg) [inline geemusic/intents/selection.py]: fix spacing

File: geemusic/utils/music.py [hunk 9]
```diff
@@ -23,6 +23,14 @@ def __init__(self, username, password, logger=None):
         )
         self.indexing_thread.start()
 
+    def populate_library(self): #TODO: Use this as a function to refresh the library with Alexa via voice commands.
```
Comment (stevenleeg) [inline geemusic/utils/music.py]: is this function used anywhere? if not let's remove it.

File: geemusic/utils/music.py [hunk 10]
```diff
@@ -151,7 +194,31 @@ def get_stream_url(self, song_id):
         return "%s/alexa/stream/%s" % (environ['APP_URL'], song_id)
 
     def get_thumbnail(self, artist_art):
-        return artist_art.replace("http://", "https://")
+        # return artist_art.replace("http://", "https://") //OLD
```
Comment (stevenleeg) [inline geemusic/utils/music.py]: remove

File: server.py [hunk 11]
```diff
@@ -1,6 +1,7 @@
 from geemusic import app
+import geemusic
 import os
 
 if __name__ == '__main__':
     port = int(os.environ.get("PORT", 4000))
-    app.run(host='0.0.0.0', port=port, debug=True)
+    app.run(host='0.0.0.0', port=port, debug=True)
```
Comment (stevenleeg) [inline server.py]: remove trailing whitespace

File: geemusic/__init__.py (hunk 3 above)
Comment (fergyfresh) [reply geemusic/__init__.py]: I ran `find . -type f -exec sed -i.orig 's/\t/    /g' {} +` everywhere and this was the only file with tabs still in it. Feel free to correct me if I'm wrong

File: geemusic/utils/music.py (hunk 9 above)
Comment (fergyfresh) [reply geemusic/utils/music.py]: nope. deleted it

**Comment by fergyfresh** [comment]
Fixed a bunch of stuff but haven't been able to test it out yet. I think I satisfied all of your comments one by one. I haven't responded to all of them though.

**Comment by fergyfresh** [comment]
Didn't work for me either. All my changes from the original @TheModdersDen PR were mainly cosmetic. I'm gonna try the original one and see if that works.

As an aside though, we should really take the thumbnails out entirely. It's causing more issues than its worth IMHO and I don't think anyone is using this on an Alexa Show in the near future anyway.

**Comment by digiltd** [comment]
Thumbnails seem to work fine for me, it would be nice to keep them, what problems are they causing?

Free music play should be possible, though some different thinking might need to be applied. The Squeezebox (logitech media server) google music plugin https://github.com/squeezebox-googlemusic/squeezebox-googlemusic works fine with the same free account. I think that loads the library first and then searches that, rather than using the google api to do the searching. Though i am not 100% sure this is the case, but when playing with gmusicapi with the command line i can list my free library and then get streams from them. As always time to investigate the problem is usually a problem :)

**Comment by fergyfresh** [comment]
Well I locked myself out of my gmail account so I think tonight is the perfect time, if I don't get it though I'm gonna close this PR so no1 has to look at it anymore.

**Pull request closed without merging**